packages/field-base/test/input-mixin.test.js

```diff
@@ -190,38 +190,63 @@ const runTests = (defineHelper, baseMixin) => {
 
     beforeEach(async () => {
       hasInputValueChangedSpy = sinon.spy();
+      valueChangedSpy = sinon.spy();
       element = fixtureSync(`<${tag}></${tag}>`);
       element.addEventListener('has-input-value-changed', hasInputValueChangedSpy);
+      element.addEventListener('value-changed', valueChangedSpy);
       await nextRender();
       input = document.createElement('input');
       element.appendChild(input);
       element._setInputElement(input);
       await nextFrame();
     });
 
-    it('should fire the event on input value presence change', async () => {
-      input.value = 'foo';
-      fire(input, 'input');
-      await nextFrame();
-      expect(hasInputValueChangedSpy.calledOnce).to.be.true;
-
-      hasInputValueChangedSpy.resetHistory();
-      input.value = '';
-      fire(input, 'input');
-      await nextFrame();
-      expect(hasInputValueChangedSpy.calledOnce).to.be.true;
+    describe('the field has no input', () => {
+      it('should fire the event once when entering input', async () => {
+        input.value = 'foo';
+        fire(input, 'input');
+        await nextFrame();
+        expect(hasInputValueChangedSpy.calledOnce).to.be.true;
```

Comment:
This doesn't actually assert that _hasInputValue has the correct value. I'll add a few more assertions.

Reply:
It turned out that the event doesn't provide `detail.value` which we could rely on in the tests. Adding it might be considered a feature, so maybe the current setup is enough.